### DISCUSSION TITLE
Ablating Boundary Condition

### DISCUSSION THREAD

**Discussion by mflattery** [idaholab/moose#17085] [Q&A Modules: General] [answered]
Hey all, 

I am new to moose and trying to implement an ablating boundary condition.  I made some adjustments to the coupled convective heating boundary condition which is working well but I need to implement a receding boundary that is dependent on the convective heating solution.  Are there examples of a melting boundary condition I could look at?  It looks like people typically implement an elastic strain type BC to model this? Are there other options?  Any information/resources are appreciated.  Thanks.

**Comment by hugary1995** [comment, marked as answer]
Hi,

Sorry for the late reply. Your question is actually very timely -- we are right now developing new capabilities to model moving interfaces/boundaries. You'll soon be able to represent the moving interface by an element subdomain modifier and a geometric cut userobject (using levelset and XFEM). It is up to you how you want to update the levelset. 

We can probably provide more help if you could let us know how your ablation boundary is receding, e.g. do you want to prescribe the movement of the boundary, or solve the movement so that it satisfies certain KKT condition on the interface?

**Reply by mflattery** [reply]
Ah cool thanks for your help!

The movement of the boundary would be dependent on the heat transfer solution.  My plan for the heat transfer piece was to modify the convective heating boundary condition to include heat flux components that account for enthalpy required to change the phase of the solid material, and other fluxes.  Once a surface temperature solution is found that balances all the components, I would have a solution for the mass flux of ablating material at each qp.  Using a material density I could get a recession velocity for the surface, which I could prescribe to the boundary.  That is my preliminary plan to try, which might change as I learn more about all the pieces here. I have looked at the level set and XFEM modules a little bit, but don't have a lot of experience with them.

**Reply by hugary1995** [reply]
Thanks for the clarification! I think @jiangwen84 is working on a very similar problem. He is using a separate embedded mesh to represent the interface. In your case, you will need to use your computed velocity to advect that embedded mesh. But unfortunately that is still under the works. Hopefully we can add this new capability soon.

**Reply by shin-chan-cpu** [reply]
Do you have a preliminary case for this problem now? I am also studying a similar problem and need help very much.
